miden-crypto/src/word.rs

```diff
@@ -0,0 +1,614 @@
+use alloc::string::String;
+use core::{
+    cmp::Ordering,
+    fmt::Display,
+    hash::{Hash, Hasher},
+    ops::Deref,
+    slice,
+};
+
+use thiserror::Error;
+use winter_crypto::Digest;
+
+const WORD_SIZE_FELT: usize = 4;
+const WORD_SIZE_BYTES: usize = 32;
+
+use super::{Felt, StarkField, ZERO};
+use crate::{
+    rand::Randomizable,
+    utils::{
+        ByteReader, ByteWriter, Deserializable, DeserializationError, HexParseError, Serializable,
+        bytes_to_hex_string, hex_to_bytes,
+    },
+};
+
+// WORD
+// ================================================================================================
+
+/// A unit of data consisting of 4 field elements.
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "&str"))]
+pub struct Word([Felt; WORD_SIZE_FELT]);
+
+impl Word {
+    /// The serialized size of the word in bytes.
+    pub const SERIALIZED_SIZE: usize = WORD_SIZE_BYTES;
+
+    /// Creates a new [Word] from the given field elements.
+    pub const fn new(value: [Felt; WORD_SIZE_FELT]) -> Self {
+        Self(value)
+    }
+
+    /// Returns the word as a slice of field elements.
+    pub fn as_elements(&self) -> &[Felt] {
+        self.as_ref()
+    }
+
+    /// Returns the word as a byte array.
+    pub fn as_bytes(&self) -> [u8; WORD_SIZE_BYTES] {
+        <Self as Digest>::as_bytes(self)
+    }
```

Comment:
nit: I would maybe put the actual implementation of `as_bytes()` here and call this method from the `Digest` impl.

Reply:
Yep, done